### DISCUSSION TITLE
Using "{Model_config = {"extra": "forbid"}" in headers causes 422 unprocessable entity error.

### DISCUSSION THREAD

**Discussion by jaykim97** [fastapi/fastapi#12998] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated
import logging
from fastapi import FastAPI, Header
from pydantic import BaseModel

# Configure logging
logging.basicConfig(
    level=logging.INFO
    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"  # Log format
)

app = FastAPI()


class CommonHeaders(BaseModel):
    model_config = {"extra": "forbid"}

    host: str
    save_data: bool
    if_modified_since: str | None = None
    traceparent: str | None = None
    x_tag: list[str] = []

@app.get("/items/")
async def read_items(headers: Annotated[CommonHeaders, Header()]):
    logging.info(f"Received headers: {headers}")
    return headers
```

### Description

I'm following [Response Model](https://fastapi.tiangolo.com/tutorial/response-model/) tutorial and came across the following error.

```
class CommonHeaders(BaseModel):
    model_config = {"extra": "forbid"}

    host: str
    save_data: bool
    if_modified_since: str | None = None
    traceparent: str | None = None
    x_tag: list[str] = []
```

Once I add `mode_config = {"extra": "forbid"}`, making curl request will return "422 Unprocessable Entity" with following response 

```
{"detail":[{"type":"extra_forbidden","loc":["header","user-agent"],"msg":"Extra inputs are not permitted","input":"curl/8.7.1"},{"type":"extra_forbidden","loc":["header","accept"],"msg":"Extra inputs are not permitted","input":"*/*"}]}
```

Following is the curl request I used which worked without the `mode_config = {"extra": "forbid"}`

```
curl -X GET "http://127.0.0.1:8000/items/" \
-H "host: example.com" \
-H "save-data: true" \
-H "if-modified-since: Mon, 27 Nov 2024 12:00:00 GMT" \
-H "traceparent: 00-abcdef1234567890abcdef1234567890-abcdef1234567890-01" \
-H "x-tag: tag1" \
-H "x-tag: tag2"
```
According to the log following is the header received by the route:
```
Received headers: host='example.com' save_data=True if_modified_since='Mon, 27 Nov 2024 12:00:00 GMT' traceparent='00-abcdef1234567890abcdef1234567890-abcdef1234567890-01' x_tag=['tag1', 'tag2']
```

At least from the log it does not appear any extra header was given. 
What could've I missed that is causing the error?

### Operating System

macOS

### Operating System Details

Sequoia 15.0.1

### FastAPI Version

0.115.5

### Pydantic Version

2.10.1

### Python Version

3.10.15

**Comment by luzzodev** [comment, marked as answer]
curl add by default some headers and you can of course override it.

Check here how to do:
https://everything.curl.dev/http/modify/headers.html

**Reply by jaykim97** [reply]
Checking on curl never came across my mind. 
Adding following headers to the request solved the issue. Thank you.
```
-H "Accept:" \
-H "User-Agent:" \
```
